### ISSUE TITLE
Running regression tests causes output spewing into the working directory

### ISSUE THREAD

**Issue by benjaminhwilliams** [xia2/xia2#271]
Two or three different regression tests seem to be chucking output into the working directory.  The output files created are (in approximately chronological order):

- `LogFiles/CRYST1_WAVE1_SWEEP1_merging-statistics.json`
- `LogFiles/CRYST1_WAVE1_SWEEP1_merging-statistics.txt`
- `LogFiles/absorption_surface.png`
- `DataFiles/xia2.cif`
- `DataFiles/xia2.mmcif`

@Anthchirp suggested this may be a race condition, with a change of directory getting left behind, however it seems to happen even with no `-n` flag.  To reproduce, do
`libtbx.pytest --regression $xia2_directory`

**Comment by benjaminhwilliams** [comment]
I can't see any cases of tests using the fixture `regression_data` without also using `run_in_tmpdir`.

**Comment by Anthchirp** [comment]
reminds me of https://github.com/xia2/xia2/issues/214

**Comment by Anthchirp** [comment]
Tried to reproduce this locally but failed.

Then again we have made some changes since January. Small molecule tests have moved away from ```run_in_tmpdir``` and instead use the procrunner parameter to set the working directory:

https://github.com/xia2/xia2/blob/1286d01bb96caeeb9df8901b4b224053770aa5c8/Test/regression/test_small_molecule.py#L20
 This way only the subprocess is affected. We should probably do that with the other tests as well and then the issue might just go away.
